pkg/notify/notify.go

```diff
@@ -0,0 +1,87 @@
+package notify
+
+import (
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/pkg/entity"
+	"github.com/jinzhu/gorm"
+	"github.com/sirupsen/logrus"
+)
+
+// A Notifier notifies about flags state
+// It returns an error if unsuccessful
+type Notifier interface {
+	Notify(*entity.Flag, itemAction, itemType) error
```

Comment:
I also want to know your preference of what to include in the `Notify` interface. For example, what do you think of 
- including the diff of current vs previous flag_snapshot? Sometimes, the flag itself can include many many segments, but the diff can be small enough and easy to visualize in Slack.
- including `updated_by`?

Reply:
I think `updated_by` makes a lot of sense to provide it. About the diff I would say it makes sense too so what do you think on providing all of the data so notifiers can make us of it as they wish?